obspy/io/gse2/bulletin.py

```diff
@@ -0,0 +1,1071 @@
+        # Put the parent id as prefix
+        # Example: smi:local/origin/351412/arrival/6389611
+        #          |        prefix        |     ident    |
+        if parent:
+            prefix = parent.resource_id.id
+        elif parent_res_id:
+            prefix = parent_res_id.id
+
+        public_id = "%s/%s" % (prefix, ident)
+        return ResourceIdentifier(public_id)
+
+    def _comment(self, text):
+        comment = Comment()
+        comment.text = text
+        comment.resource_id = ResourceIdentifier(prefix=self.res_id_prefix)
+        return comment
+
+    def _get_creation_info(self):
+        creation_info = CreationInfo(creation_time=UTCDateTime())
+        if self.agency:
+            creation_info.agency_id = self.agency
+        if self.author:
+            creation_info.author = self.author
+        return creation_info
+
+    def _check_header(self, first_line):
+        """
+        Just check some stuff in header.
+
+        :type first_line: str
+        :param first_line: First line of header.
+        """
+        line_1_pattern = 'BEGIN\sGSE2.0'
+        line_2_pattern = 'MSG_TYPE\s(REQUEST|DATA|SUBSCRIPTION|PROBLEM)'
+        line_3_pattern = 'MSG_ID\s\w{1,20}\s?\w{1,8}'
+
+        if not re.match(line_1_pattern, first_line):
+            raise SyntaxError('Wrong GSE2.0 header')
+        if not re.match(line_2_pattern, next(self.lines)):
+            raise SyntaxError('Wrong message type in header')
+        if not re.match(line_3_pattern, next(self.lines)):
+            raise SyntaxError('Wrong message ID in header')
+
+    def _get_channel(self, station, time):
+        """
+        Use inventory to retrieve channel and location code.
+
+        :type station: str
+        :param station: Station code
+        :type time: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param time: Only use channel active at given point in time.
+        :rtype: str, :class:`~obspy.core.inventory.channel.Channel`
+        :return: Network code, channel retrieved.
+        """
+        if self.inventory is None:
+            return None
+
+        sub_inventory = self.inventory.select(
+                        station=station,
+                        time=time)
+
+        if len(sub_inventory) == 0:
+            self._warn("Can't find station %s in inventory" % station)
+            return None
+
+        if len(sub_inventory) > 1:
+            self._warn('Several stations %s have been found with different '
+                       'network code in inventory' %
+                       station)
+            return None
+
+        network = sub_inventory[0]
+
+        sta = [cha for sta in network for cha in sta]
+
+        # Select only vertical channels
+        channels = [cha for cha in sta if cha.code[-1] == 'Z']
+
+        if len(channels) == 0:
+            self._warn("Can't find channel for station %s" % station)
+            return network.code, None
+
+        if len(channels) > 1:
```

Comment:
The whole block beneath this `if` statement is not tested according to codecov. As this is a non-trivial piece of code, please add a test for this.

In case you don't know about this - its really helpful: https://github.com/codecov/browser-extension (once the continuous integration has been run you'll be able to see test coverage for pull requests right in the browser).